pkg/peertls/extensions/extensions.go

```diff
@@ -127,8 +127,11 @@ func AddSignedCert(key crypto.PrivateKey, cert *x509.Certificate) error {
 	return nil
 }
 
-// AddExtension adds one or more extensions to a certificate
-func AddExtension(cert *x509.Certificate, exts ...pkix.Extension) (err error) {
+// AddExtraExtension adds one or more extensions to a certificate for serialization.
+// NB: this *does not* serialize or persist the extension into the certificates's
+// raw bytes. To add a persistent extension use `FullCertificateAuthority.AddExtension`
```

Comment:
when would a not persistent extension be used over a persistent one?

Reply:
Probably never but in order for an extension to be serialized (i.e. in `Extensions` when deserialized), it first has to be added to `ExtraExtensions`.

Example:
```golang
originalCert := x509.CreateCertificate(...)
originalCert.ExtraExtensions = append(originalCert.ExtraExtensions, pkix.Extension{...})

// originalCert is passed as `template` so that its `ExtraExtensions` are read
extendedCertBytes, err := x509.CreateCertificate(rand.Reader, originalCert, ...)

// when parsed, `extendedCert.Extensions` will contain the extension and `extendedCert.ExtraExtensions` will be empty.
extendedCert, err = x509.ParseCertificate(extendedCertBytes)
```